tests/queries/0_stateless/03164_optimize_row_order_during_insert.sql

```diff
@@ -0,0 +1,10 @@
+-- Checks that no bad things happen when the table optimizes the row order to improve compressability during just simple insert.
+
+DROP TABLE IF EXISTS tab;
+
+CREATE TABLE tab (name String, event Int8) ENGINE = MergeTree ORDER BY name SETTINGS allow_experimental_optimized_row_order = true;
+INSERT INTO tab VALUES ('Igor', 3), ('Egor', 1), ('Egor', 2), ('Igor', 2), ('Igor', 1);
+
+SELECT * FROM tab SETTINGS max_threads=1;
```

Comment:
The SELECT has to have an ORDER BY, otherwise the database is in principle free to return the result in arbitrary order.

The trick is to understand that
- with ORDER BY, and
- with the ORDER BY columns forming a prefix of the table's primary key, and
- with max_threads = 1

the "read-in-order" optimization kicks in. This makes sure that the ORDER BY operator is removed from the query plan and the content of the data part is returned verbatim, i.e. without additional sorting.

Reply:
Yes: `SELECT * FROM tab ORDER BY name SETTINGS max_threads=1;`

Reply:
I have corrected the tests according to your comments.